README.md

```diff
@@ -177,12 +177,16 @@ Install Dex:
 kustomize build common/dex/overlays/istio | kubectl apply -f -
 ```
 
-#### OIDC AuthService
+#### Authservice
 
-The OIDC AuthService extends your Istio Ingress-Gateway capabilities, to be able to function as an OIDC client:
+The Authservice is responsible for ensuring requests are authorized. Otherwise redirecting them to the provider (Here Dex is used as OIDC Provider). It is also responsible for adding the `kubeflow-userid` to the upstream request.
+
+The [OAuth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy) will extends the istio Ingress-Gateway in the same way the *OIDC AuthService* is doing but offer more flexibility for handling requests. See the [documentation](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview).
+
+Install oauth2-proxy
 
 ```sh
-kustomize build common/oidc-authservice/base | kubectl apply -f -
+kustomize build common/auth-proxy/overlays/oauth-proxy | kubectl apply -f -
```

Comment:
Let's leave the `oidc-authservice` there by default for this release and we can make the migration in the next release to only oauth2-proxy.

But we could do something like this there:
```sh
kustomize build common/oidc-authservice/base | kubectl apply -f -
# kustomize build common/auth-proxy/overlays/oauth-proxy | kubectl apply -f -
```